ibis/expr/operations/relations.py

```diff
@@ -322,7 +309,7 @@ def root_tables(self):
 class SelfReference(TableNode, sch.HasSchema):
     table = rlz.table
 
-    @cached_property
+    @property
```

Comment:
Why remove the caching?

Reply:
Cached property can add more overhead if `self.table.schema()` is cheap. I think we should use cached properties only where schema is actually calculated.

Reply:
Or even better, materialize the expensive properties in `__init__()` since the nodes are immutable now.